lib/netext/httpext/transport.go

```diff
@@ -115,9 +118,26 @@ func (t *transport) measureAndEmitMetrics(unfReq *unfinishedRequest) *finishedRe
 			tags["status"] = "0"
 		}
 	} else {
+		cleanURL := URL{u: unfReq.request.URL, URL: unfReq.request.URL.String()}.Clean()
 		if enabledTags.Has(stats.TagURL) {
-			u := URL{u: unfReq.request.URL, URL: unfReq.request.URL.String()}
-			tags["url"] = u.Clean()
+			tags["url"] = cleanURL
+		}
+
+		if _, ok := tags["name"]; !ok && enabledTags.Has(stats.TagName) {
+			// If a name tag is already set in the script, use it for all requests
+			// processed by this transport. Otherwise, use the original URL name
+			// to preserve any escaped query string arguments, but only for the
+			// first request in the chain.
+			if t.origURLName != "" {
+				tags["name"] = t.origURLName
+				t.origURLName = ""
```

Comment:
This still seems strange... if a `name` tag was set by the user in the script, wouldn't it be in the `t.tags` map above? :confused:

Reply:
ah, nevermind, "spoke" too soon again - this comment is about the `if _, ok := tags["name"]; !ok` part of the check...

still, I continue to not understand why we need the original URL, when `cleanURL` should suffice :confused:

Reply:
Because otherwise [this test](https://github.com/loadimpact/k6/blob/5ff4a76a4a08241ce061fcc449be585f7f707e6a/js/modules/k6/http/request_test.go#L979) fails.

I'm trying to see if I can clean it again in the transport the same way while preserving the escaped query args, but I'm not functioning on all cylinders today. :-/

Reply:
> Because otherwise [this test](https://github.com/loadimpact/k6/blob/5ff4a76a4a08241ce061fcc449be585f7f707e6a/js/modules/k6/http/request_test.go#L979) fails.

Ahhh... welll... :hankey: :sweat_smile: Yeah, I'll have to scratch my head about this some more, but the current version of this PR might be as good as it gets... :disappointed: Not sure why anyone added [this](https://k6.io/docs/using-k6/http-requests#url-grouping) (on the bottom of that section) [funky](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals#Tagged_templates) way of specifying URLs, but now we have to live with it... :sob: It implies a `name` tag, but it doesn't specify it, instead using this weird k6-specific [`URL` struct](https://github.com/loadimpact/k6/blob/80d3b06a28c8612846cfcbe8783100e84015d394/lib/netext/httpext/request.go#L51-L57), which of course we can't propagate in an [`http.RoundTrippter`](https://golang.org/pkg/net/http/#RoundTripper) :sob: And thus, I finally understand the purpose of all of the weird fixes in this PR :sweat_smile: 

Let's discuss this on Monday again, but I'm thinking that if someone uses something like this:
```js
http.get(http.url`http://example.com/posts/${id}`)
```
 we probably should consider it as if they used
```
http.get(`http://example.com/posts/${id}`, {tags: {name: 'http://example.com/posts/${}'}})
```

That is, if there are any redirects in the course of that HTTP request, all of their `name` tags should be 'http://example.com/posts/${}'. Or, said otherwise, https://github.com/loadimpact/k6/issues/1474 is not a bug when the user has explicitly used the `http.url` tagged template.

Reply:
OK, I think that should be the case in 62f91845. The only wonky thing is the obscure `preq.URL.Name != preq.URL.CleanURL` check, which works when a tagged template string is used.

This does break the masked credentials behavior for digest auth, i.e. the generated `name` tag in these cases won't be `https://****:****@<host>...` anymore, but `https://<host>`. I discussed this with @na-- the other day, so this might be acceptable, and if so we should mention it in the release notes.